src/ucs/config/parser.c

```diff
@@ -347,6 +347,37 @@ int ucs_config_sprintf_enum(char *buf, size_t max,
     return 1;
 }
 
+int ucs_config_sscanf_uint_enum(const char *buf, void *dest, const void *arg)
+{
+    int i;
+
+    i = ucs_string_find_in_list(buf, (const char**)arg, 0);
+    if (i >= 0) {
+        *(unsigned*)dest = UINT_MAX - i;
+        return 1;
+    }
+
+    return sscanf(buf, "%u", (unsigned*)dest);
+}
+
+int ucs_config_sprintf_uint_enum(char *buf, size_t max,
+                                 const void *src, const void *arg)
+{
+   char* const *table = arg;
+   int table_size;
+
+   for (table_size = 0; table[table_size] != NULL; table_size++) {
+       continue;
+   }
+
+   if ((UINT_MAX - table_size) < *(unsigned*)src) {
+        strncpy(buf, table[UINT_MAX - *(unsigned*)src], max);
```

Comment:
use the macro UCS_CONFIG_UINT_ENUM_INDEX
use helper var "value = *(unsigned*)src"